crates/viewer/re_space_view_spatial/src/visualizers/videos.rs

```diff
@@ -191,49 +191,67 @@ impl VideoFrameReferenceVisualizer {
-                                world_from_entity,
-                                err.to_string(),
-                                video_resolution,
-                                entity_path,
-                            );
-                            None
-                        }
-                    }
-                {
-                    let textured_rect = TexturedRect {
-                        top_left_corner_position: world_from_entity
-                            .transform_point3(glam::Vec3::ZERO),
+
+                match video.frame_at(render_ctx, decode_stream_id, video_timestamp.as_seconds()) {
+                    Ok(frame) => {
                         // Make sure to use the video instead of texture size here,
-                        // since it may be a placeholder which doesn't have the full size yet.
-                        extent_u: world_from_entity
-                            .transform_vector3(glam::Vec3::X * video_resolution.x),
-                        extent_v: world_from_entity
-                            .transform_vector3(glam::Vec3::Y * video_resolution.y),
-                        colormapped_texture: ColormappedTexture::from_unorm_rgba(texture),
-                        options: RectangleOptions {
-                            texture_filter_magnification: TextureFilterMag::Nearest,
-                            texture_filter_minification: TextureFilterMin::Linear,
-                            outline_mask: spatial_ctx.highlight.overall,
-                            ..Default::default()
-                        },
-                    };
-                    self.data.pickable_rects.push(PickableTexturedRect {
-                        ent_path: entity_path.clone(),
-                        textured_rect,
-                        source_data: PickableRectSourceData::Video,
-                    });
+                        // since the texture may be a placeholder which doesn't have the full size yet.
+                        let top_left_corner_position =
+                            world_from_entity.transform_point3(glam::Vec3::ZERO);
+                        let extent_u =
+                            world_from_entity.transform_vector3(glam::Vec3::X * video_resolution.x);
+                        let extent_v =
+                            world_from_entity.transform_vector3(glam::Vec3::Y * video_resolution.y);
+
+                        let texture = match frame {
+                            VideoFrameTexture::Ready(texture) => texture,
+                            VideoFrameTexture::Pending(placeholder) => {
+                                // Show loading rectangle:
+                                let min = top_left_corner_position;
+                                let max = top_left_corner_position + extent_u + extent_v;
+                                let center = 0.5 * (min + max);
+                                let diameter = (max - min).truncate().abs().min_element();
```

Comment:
this likely breaks in 3D https://github.com/rerun-io/rerun/issues/7543

.. but I'm actually not even sure, gotta check how we place labels on 3D points. Likely we'll have to do a similar thing!